demos/source/aws_iot_demo_defender.c

```diff
@@ -212,35 +250,78 @@ static AwsIotDefenderError_t _defenderDemo( const char * pIdentifier,
+
+    /* Set MQTT connection information. */
+    mqttConnectionInfo.pClientIdentifier = pIdentifier;
+    mqttConnectionInfo.clientIdentifierLength = ( uint16_t ) strlen( pIdentifier );
+
+    return IotMqtt_Connect( &mqttNetworkInfo,
+                            &mqttConnectionInfo,
+                            1000,
+                            pMqttConnection );
+}
+
+/*-----------------------------------------------------------*/
+
+static void _stopMqttConnection( IotMqttConnection_t mqttConnection )
+{
+    IotMqtt_Disconnect( mqttConnection, false );
+}
+
+/*-----------------------------------------------------------*/
+
 static AwsIotDefenderError_t _startDefender( const char * pIdentifier,
                                              void * pNetworkServerInfo,
                                              void * pNetworkCredentialInfo,
                                              const IotNetworkInterface_t * pNetworkInterface )
 {
     const AwsIotDefenderCallback_t callback = { .function = _defenderCallback, .param1 = NULL };
-    AwsIotDefenderError_t defenderResult;
+    AwsIotDefenderError_t defenderResult = AWS_IOT_DEFENDER_SUCCESS;
 
     AwsIotDefenderStartInfo_t startInfo = AWS_IOT_DEFENDER_START_INFO_INITIALIZER;
+    IotMqttError_t mqttError = IOT_MQTT_SUCCESS;
 
-    /* Set network information. */
-    startInfo.mqttNetworkInfo = ( IotMqttNetworkInfo_t ) IOT_MQTT_NETWORK_INFO_INITIALIZER;
-    startInfo.mqttNetworkInfo.createNetworkConnection = true;
-    startInfo.mqttNetworkInfo.u.setup.pNetworkServerInfo = pNetworkServerInfo;
-    startInfo.mqttNetworkInfo.u.setup.pNetworkCredentialInfo = pNetworkCredentialInfo;
-
-    /* Set network interface. */
-    startInfo.mqttNetworkInfo.pNetworkInterface = pNetworkInterface;
-
-    /* Set MQTT connection information. */
-    startInfo.mqttConnectionInfo = ( IotMqttConnectInfo_t ) IOT_MQTT_CONNECT_INFO_INITIALIZER;
-    startInfo.mqttConnectionInfo.pClientIdentifier = pIdentifier;
-    startInfo.mqttConnectionInfo.clientIdentifierLength = ( uint16_t ) strlen( pIdentifier );
+    /* Set MQTT subscription information */
+    startInfo.pClientIdentifier = pIdentifier;
+    startInfo.clientIdentifierLength = ( uint16_t ) strlen( pIdentifier );
 
     /* Set Defender callback. */
     startInfo.callback = callback;
 
-    /* Invoke defender start API. */
-    defenderResult = AwsIotDefender_Start( &startInfo );
+    /* Create MQTT Connection */
+    mqttError = _startMqttConnection( pIdentifier,
+                                      pNetworkServerInfo,
+                                      pNetworkCredentialInfo,
+                                      pNetworkInterface,
+                                      &_mqttConnection );
+
+    if( mqttError != IOT_MQTT_SUCCESS )
+    {
+        IotLogError( "Failed to Create MQTT Connection" );
+        /* TODO map to defender error */
+    }
+    else
+    {
+        startInfo.mqttConnection = _mqttConnection;
```

Comment:
Parts of ```startInfo``` are initialized in different places. Can this be consolidated together so that its easier to read ?